pkg/controller/direct/compute/forwardingrule_controller.go

```diff
@@ -0,0 +1,460 @@
+		if err != nil {
+			return nil, err
+
+		}
+		obj.Spec.Target.TargetHTTPProxyRef.External = targetHTTPProxyRef.Url()
+	}
+
+	// Get target TargetVPNGateway
+	if obj.Spec.Target.TargetVPNGatewayRef != nil {
+		targetVPNGatewayRef, err := refs.ResolveComputeTargetVPNGateway(ctx, reader, obj, obj.Spec.Target.TargetVPNGatewayRef)
+		if err != nil {
+			return nil, err
+
+		}
+		obj.Spec.Target.TargetVPNGatewayRef.External = targetVPNGatewayRef.Url()
+	}
+
+	// Get location
+	location := obj.Spec.Location
+
+	forwardingRuleAdapter := &forwardingRuleAdapter{
+		resourceID: resourceID,
+		projectID:  projectID,
+		Location:   location,
+		desired:    obj,
+	}
+
+	// Get compute GCP client
+	if location == "global" {
+		gcpClient, err := m.globalForwardingRulesClient(ctx)
+		if err != nil {
+			return nil, err
+		}
+		forwardingRuleAdapter.globalForwardingRulesClient = gcpClient
+	} else {
+		gcpClient, err := m.forwardingRulesClient(ctx)
+		if err != nil {
+			return nil, err
+		}
+		forwardingRuleAdapter.forwardingRulesClient = gcpClient
+	}
+	return forwardingRuleAdapter, nil
+}
+
+func (m *forwardingRuleModel) AdapterForURL(ctx context.Context, url string) (directbase.Adapter, error) {
+	// TODO: Support URLs
+	return nil, nil
+}
+
+func (a *forwardingRuleAdapter) Find(ctx context.Context) (bool, error) {
+	if a.resourceID == "" {
+		return false, nil
+	}
+
+	log := klog.FromContext(ctx).WithName(ctrlName)
+	log.V(2).Info("getting ComputeForwardingRule", "name", a.resourceID)
+
+	var err error
+	forwardingRule := &computepb.ForwardingRule{}
+	if a.Location == "global" {
+		req := &computepb.GetGlobalForwardingRuleRequest{
+			ForwardingRule: a.resourceID,
+			Project:        a.projectID,
+		}
+		forwardingRule, err = a.globalForwardingRulesClient.Get(ctx, req)
+	} else {
+		req := &computepb.GetForwardingRuleRequest{
+			ForwardingRule: a.resourceID,
+			Region:         a.Location,
+			Project:        a.projectID,
+		}
+		forwardingRule, err = a.forwardingRulesClient.Get(ctx, req)
+	}
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting ComputeForwardingRule %q failed: %w", a.resourceID, err)
+	}
+	a.actual = forwardingRule
+	return true, nil
+}
+
+func (a *forwardingRuleAdapter) Create(ctx context.Context, u *unstructured.Unstructured) error {
+	if a.projectID == "" {
+		return fmt.Errorf("project is empty")
+	}
+	if a.resourceID == "" {
+		return fmt.Errorf("resourceID is empty")
+	}
+
+	log := klog.FromContext(ctx).WithName(ctrlName)
+	log.V(2).Info("creating ComputeForwardingRule", "name", a.resourceID)
+	mapCtx := &direct.MapContext{}
+
+	desired := a.desired.DeepCopy()
+
+	forwardingRule := ComputeForwardingRuleSpec_ToProto(mapCtx, &desired.Spec)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	forwardingRule.Name = direct.LazyPtr(a.resourceID)
+	forwardingRule.Labels = desired.Labels
+	// TODO(yuhou): handle default spec values
```

Comment:
Who defaults the LoadBalancingScheme value today?  Terraform?

Reply:
It can be provided by user, if not, then TF default it to "EXTERNAL"([TF code](https://source.corp.google.com/piper///depot/google3/third_party/golang/terraform_providers/google_beta/google_beta/services/compute/resource_compute_forwarding_rule.go;l=232?q=resource_compute_forwarding_rule.go&sq=package:piper%20file:%2F%2Fdepot%2Fgoogle3%20-file:google3%2Fexperimental))

Reply:
I think we have two questions:
1. What does the GCP service do if not providing the default value (i.e. does it use a different default)? 
2. I assume TF does this for a "good" reason. But we want to know the reason first to decide whether or not use the same behavior. And if you do not know why TF does it, you can assume you know nothing about this special handling. (I doubt if this is anything related to `state-into-spec`)